### PR TITLE
adds daemon flag for env var and turbo.json

### PR DESCRIPTION
**Pull request by dimitropoulos**
### Description

This PR adds support and testing for a `TURBO_DAEMON` environment variable, as well as a `{ daemon: boolean }` field to the `turbo.json` file.  This change allows us to remove `daemon` from `RunOpts`, as well.

### Testing Instructions

You can run `turbo daemon status` to see the status of the daemon after each run (and also `turbo daemon stop` to kill it).

#### Daemon Off

1. run with `--no-daemon`
1. run with `TURBO_DAEMON=false`
1. run with `{"daemon":false}` in your turbo.json

#### Daemon On

1. run with `--daemon`
1. run with `TURBO_DAEMON=true`
1. run with `{"daemon":true}` in your turbo.json

### CODE REVIEW

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 7a9e23b75961d269eb99f86685b01b17645f17fe

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Review by mehulkar — APPROVED** [review]
Does `--no-daemon` still work also?

File: crates/turborepo-lib/src/turbo_json/mod.rs
Comment (chris-olszewski) [inline crates/turborepo-lib/src/turbo_json/mod.rs:1083]: So something I missed: when we add new keys/fields to `turbo.json` we should always test both deserialization and serialization. When we don't we often create issues ([most recent issue caused by me forgetting to test this](https://github.com/vercel/turbo/issues/8311))

Comment (dimitropoulos) [reply crates/turborepo-lib/src/turbo_json/mod.rs]: sounds good! updated in https://github.com/vercel/turbo/pull/8728/commits/9065d956ad5d42abb1fd9744a98a2be105b01b9a

**Comment by dimitropoulos** [comment]
@mehulkar yep :) `--no-daemon` is unchanged (still works).  there was an error in the description (updated now).

File: crates/turborepo-lib/src/run/builder.rs
Comment (NicholasLYang) [inline crates/turborepo-lib/src/run/builder.rs:58]: Could we keep this in `Opts` but use the config value to set `daemon`? That way we can keep using `Opts` as the single source of truth

**Review by NicholasLYang — CHANGES_REQUESTED** [review]
I think we should keep this config in `Opts` and not in `RunBuilder`/`Run` directly

File: crates/turborepo-lib/src/run/builder.rs
Comment (dimitropoulos) [reply crates/turborepo-lib/src/run/builder.rs]: looks like we're all set here! see https://github.com/vercel/turbo/pull/8728/commits/60e9a54dd1c002583d78f6b9f5dd02b2797e541e

File: turborepo-tests/integration/tests/config.t
Comment (NicholasLYang) [inline turborepo-tests/integration/tests/config.t:19]: Might be good to add a test that confirms the overloading behavior with an env var or flag.

**Review by NicholasLYang — APPROVED** [review]
Needs a test, but otherwise LGTM!

File: turborepo-tests/integration/tests/config.t
Comment (dimitropoulos) [reply turborepo-tests/integration/tests/config.t:19]: thanks! updated in a7216fad72a4bd19d34519d10a29677e9cee3671

**Comment by shawnmcknight** [comment]
@dimitropoulos Is there something separate that needs to be done to get the json schema for `turbo.json` updated? The `daemon` property does not appear to be present in https://turbo.build/schema.json.

**Comment by PatrykKuniczak** [comment]
@dimitropoulos How to use this 
``` {"daemon":false} ```
in turbo.json?

I'm using latest version but i can't do it, also there's no DOCS for this

**Comment by dimitropoulos** [comment]
@shawnmcknight my understanding is that there's a separate process (in another (private) repo) that generates that schema.  I'm looking into why it's not been updated - sorry about that!

**Comment by dimitropoulos** [comment]
@PatrykKuniczak we're following up with the docs for this and the other two, [cacheDir](github.com/vercel/turbo/pull/8947) (in progress) and [envMode](https://github.com/vercel/turbo/pull/8757) (complete) at the same time.

**Pull request merged**
